test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStringLatin1.java

```diff
@@ -0,0 +1,163 @@
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package jit.test.recognizedMethod;
+import org.testng.AssertJUnit;
+import org.testng.annotations.Test;
+
+public class TestJavaLangStringLatin1 {
+
+    @Test(groups = {"level.sanity"}, invocationCount=2)
+    public void test_java_lang_StringLatin1_inflate() {
+        char[] expected = new char[0];
+        char[] actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 0 array", expected, actual);
+
+        expected = new char[] { '\u0054' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 1 array", expected, actual);
+
+        expected = new char[] { '\u0032', '\u0070' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 2 array", expected, actual);
+
+        expected = new char[] { '\u0076', '\u0065', '\u0084' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 3 array", expected, actual);
+
+        expected = new char[] { '\u00AF', '\u00A6', '\u0084', '\u005F' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 4 array", expected, actual);
+
+        expected = new char[] { '\u0025', '\u002C', '\u0068', '\u0087', '\u0077' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 5 array", expected, actual);
+
+        expected = new char[] { '\u00A0', '\u0033', '\u0052', '\u005D', '\u0022', '\u0098' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 6 array", expected, actual);
+
+        expected = new char[] { '\u009B', '\u0076', '\u007B', '\u0040', '\u003A', '\u003B', '\u00A7' };
+        actual = new String(expected).toCharArray();
+        AssertJUnit.assertArrayEquals("Incorrect result for length 7 array", expected, actual);
+
+        expected = new char[] { '\u0052', '\u00A2', '\u0040', '\u0035', '\u006E', '\u0052', '\u0074', '\u008D' };
```

Comment:
During debugging (while working on inlining `StringLatin1.inflate`), I mainly used input strings that looked like below:
```
{ '\u0052', '\u0053', '\u0054', '\u0055', '\u0056', '\u0057', '\u0058', '\u0059' };
```

If the characters are random, then it required a little more manual scanning to find where an issue is. (ex. we might miscalculate an offset and hence overwrite some data starting at an earlier point in the `actual` array)

Obviously with automated testing the character value doesn't matter. However if we run into edge case failures once the inlining code is merged, then having a predictable sequence of characters in these array could make debugging easier as it can be easier to take a quick glance at a memory and figure out where we start corrupting the data. Given that do you think it'd be better to have a more predictable sequence of characters in the input strings?

Reply:
That's a great point. I totally agree. Fixed in [force-push](https://github.com/eclipse/openj9/compare/780a7d712a5d251f5bcfeb3b5a1451e10a1c79c0..d3df740fd2c2d6699aa531e035b481f7673e1a5c).